certification/pyxis/builder.go

```diff
@@ -150,43 +134,38 @@ func (b *certificationInputBuilder) storeRPMManifest(filepath string) error {
 		)
 	}
 
-	var manifest RPMManifest
-	err = json.Unmarshal(bytes, &manifest)
+	err = json.Unmarshal(bytes, &submission)
 	if err != nil {
 		return fmt.Errorf(
-			"%w: data for the %s appears to be malformed: %s",
+			"%w: json data appears to be malformed: %s",
 			errors.ErrSubmittingToPyxis,
-			"rpm manifest",
 			err,
 		)
 	}
 
+	return nil
+}
+
+// storeRPMManifest reads the manifest from disk at path and stores it in
+// the CertificationInput as an RPMManifest struct.
+func (b *certificationInputBuilder) storeRPMManifest(filepath string) error {
+	var manifest RPMManifest
+	err := readAndUnmarshal(filepath, &manifest)
+	if err != nil {
+		return err
+	}
+
 	b.RpmManifest = &manifest
 	return nil
 }
```

Comment:
I'm not sure If I wanted to remove the string representing the asset we were working with here. The only reason I like it is because we do this same kind of logic multiple times, but with different files. In other words, there's a high likelihood that we'll see similar error messages if any of the instances that perform this kind of logic fail, so differentiating might be difficult.

Did you see somewhere else where we would be giving that little bit of context to a debugging developer that justifies not having it here?

Reply:
Perhaps we should have the more elaborate error being creating in each of the store methods, rather than in the extracted logic. They would have that context. Or, we just need to add the context here... Let me play with that a bit.

Reply:
Let me know what you think of the solution in the latest commit.

Reply:
Hey! That works nicely. 👍